src/compiler/emitter.ts

```diff
@@ -5546,15 +5542,16 @@ namespace ts {
         }
 
         function emitLeadingComment(commentPos: number, commentEnd: number, kind: SyntaxKind, hasTrailingNewLine: boolean, rangePos: number) {
-            if (!shouldWriteComment(currentSourceFile!.text, commentPos)) return;
+            Debug.assertIsDefined(currentSourceFile);
+            if (!shouldWriteComment(currentSourceFile.text, commentPos)) return;

```

Comment:
No real need to assert here, we can just exit early.
```suggestion
            if (!currentSourceFile || !shouldWriteComment(currentSourceFile.text, commentPos)) return;
```